code/src/java/pcgen/util/Comparators.java

```diff
@@ -17,139 +17,66 @@
-	private static final Comparator<Object> tSICC = new ToStringIgnoreCaseComparator();
-	private static final Comparator<Object> tSICCol = new ToStringIgnoreCaseCollator();
-	private static final Comparator<Integer> iC = new IntegerComparator();
-	private static final Comparator<Number> nC = new NumberComparator();
-	private static final Comparator<Date> DC = new DateComparator();
-	private static final Comparator<Object> treeNodeComp = new TreeTableNodeComparator();
-
-	private static Comparator<Object> toStringComparator()
-	{
-		return tSC;
-	}
-
-	public static Comparator<Object> toStringIgnoreCaseComparator()
-	{
-		return tSICC;
-	}
-
-	public static Comparator<Object> toStringIgnoreCaseCollator()
-	{
-		return tSICCol;
-	}
-
-	/**
-	 * @return A comparator for use with the contents of tree table nodes. 
-	 */
-	private static Comparator<Object> treeTableNodeComparator()
-	{
-		return treeNodeComp;
-	}
-
-	private static Comparator<Integer> integerComparator()
-	{
-		return iC;
-	}
-
-	private static Comparator<Number> numberComparator()
-	{
-		return nC;
-	}
-
-	private static Comparator<Date> dateComparator()
-	{
-		return DC;
-	}
-
-	private static Comparator<String> ignoreCaseStringComparator()
-	{
-		return String.CASE_INSENSITIVE_ORDER;
-	}
+	private static final Comparator<Object> toStringComparator = new ToStringComparator();
+	private static final Comparator<Object> treeTableNodeComparator = new TreeTableNodeComparator();
 
 	public static <T> Comparator<? super T> getComparatorFor(Class<T> c)
 	{
-		if (c == Integer.class)
-		{
-			return (Comparator<? super T>) integerComparator();
-		}
-		else if (c.getSuperclass() == Number.class)
+		if (c.getSuperclass() == Number.class)
 		{
-			return (Comparator<? super T>) numberComparator();
+			return (Comparator<? super T>) Comparator.comparing(Function.identity());
 		}
 		else if (c == Date.class)
 		{
-			return (Comparator<? super T>) dateComparator();
+			return (Comparator<? super T>) Comparator.comparingLong(Date::getTime);
 		}
 		else if (c == String.class)
 		{
-			return (Comparator<? super T>) ignoreCaseStringComparator();
+			return (Comparator<? super T>) String.CASE_INSENSITIVE_ORDER;
 		}
 		else if (c == TreeTableNode.class || c == InfoFacade.class || c.getSuperclass() == InfoFacade.class)
 		{
-			return treeTableNodeComparator();
+			return treeTableNodeComparator;
 		}
-		return toStringComparator();
+		return toStringComparator;
 	}
 
 	/**
 	 * A {@code Comparator} to compare objects as
 	 * {@code String}s.  This is particularly useful for applications
 	 * such as maintaining a sorted {@code JComboBoxEx} and the like.
 	 */
-	private static final class ToStringComparator implements Comparator<Object>, Serializable
+	private static final class ToStringComparator implements Comparator<Object>
 	{
-
 		@Override
 		public int compare(Object o1, Object o2)
 		{
-			// Treat null as the empty string.
```

Comment:
I like this because it shows developer intent and is not the traditional way a compare would work